app/src/main/java/com/github/kulebin/myfoursquareapp/adapter/VenueListAdapter.java

```diff
@@ -39,17 +46,30 @@ public int getItemCount() {
 
     public static class VenueItemViewHolder extends RecyclerView.ViewHolder implements VenueItemView {
 
+        String venueId;
+        @BindView(R.id.text_venue_name)
         TextView nameView;
+        @BindView(R.id.text_venue_location)
         TextView addressView;
+        @BindView(R.id.text_venue_rating)
         TextView ratingView;
+        @BindView(R.id.image_venue)
         ImageView imageView;
+        VenueListContract.Presentation.OnItemListener itemListener;
 
         public VenueItemViewHolder(final View itemView) {
             super(itemView);
-            this.nameView = (TextView) itemView.findViewById(R.id.venueNameTextView);
-            this.addressView = (TextView) itemView.findViewById(R.id.venueLocationTextView);
-            this.ratingView = (TextView) itemView.findViewById(R.id.venueRatingTextViewView);
-            this.imageView = (ImageView) itemView.findViewById(R.id.venueImageView);
+
+            ButterKnife.bind(this, itemView);
+            itemView.setOnClickListener(new View.OnClickListener() {
+
```

Comment:
Don't recreate ClickListeners for every call, you can set it directly in com.github.kulebin.myfoursquareapp.adapter.VenueListAdapter.VenueItemViewHolder#setOnItemListener method (just make viewGroup a field to have access), you can create OnClickListener in VenueListPresenter for this